src/audiomixerboard.cpp

```diff
@@ -1210,92 +1210,98 @@ void CAudioMixerBoard::ApplyNewConClientList ( CVector<CChannelInfo>& vecChanInf
 
         // search for channels with are already present and preserve their gain
         // setting, for all other channels reset gain
-        for ( int i = 0; i < MAX_NUM_CHANNELS; i++ )
+
+        // get all channels which are in use/not in use. We use the array index of vecChanInfo
+        // if the fader is in use, else INVALID_INDEX to specify it is not in use
+        int iFaderNumber[MAX_NUM_CHANNELS];
+
+        for ( int iChanID = 0; iChanID < MAX_NUM_CHANNELS; iChanID++ )
+        {
+            iFaderNumber[iChanID] = INVALID_INDEX;
+        }
+
+        for ( int iFader = 0; iFader < iNumConnectedClients; iFader++ )
         {
-            bool bFaderIsUsed = false;
+            iFaderNumber[vecChanInfo[iFader].iChanID] = iFader;
+        }
+
+        for ( int iChanID = 0; iChanID < MAX_NUM_CHANNELS; iChanID++ )
+        {
+            if ( iFaderNumber[iChanID] == INVALID_INDEX )
+            {
+                // fader is not used --> hide it
+                // before hiding the fader, store its level (if some conditions are fulfilled)
+                StoreFaderSettings ( vecpChanFader[iChanID] );
 
-            for ( int j = 0; j < iNumConnectedClients; j++ )
+                vecpChanFader[iChanID]->Hide();
+                continue;
+            }
+
+            // current fader is used
+            // check if fader was already in use -> preserve gain value
+            if ( !vecpChanFader[iChanID]->IsVisible() )
             {
-                // check if current fader is used
-                if ( vecChanInfo[j].iChanID == i )
+                // the fader was not in use, reset everything for new client
+                vecpChanFader[iChanID]->Reset();
+                vecAvgLevels[iChanID] = 0.0f;
+
+                // check if this is my own fader and set fader property
```

Comment:
Don't say "check if" on an if statement.  That's what an if statement _is_.  Inside the if -- if the test doesn't make it clear -- explain the state.  Here, a comment of "// this is my own fader" would be present.  However, it's pretty redundant.